pkg/sql/schemachanger/scbuild/internal/scbuildstmt/helpers.go

```diff
@@ -292,6 +292,41 @@ func getSortedColumnIDsInIndex(
 	return keyColumnIDs, keySuffixColumnIDs, storingColumnIDs
 }
 
+// getNonDropColumnIDs returns all non-drop column IDs in a table in ascending order.
+func getNonDropColumnIDs(b BuildCtx, tableID catid.DescID) (ret []catid.ColumnID) {
+	scpb.ForEachColumn(b.QueryByID(tableID).Filter(publicTargetFilter), func(
+		_ scpb.Status, _ scpb.TargetStatus, e *scpb.Column,
+	) {
+		ret = append(ret, e.ColumnID)
+	})
+	sort.Slice(ret, func(i, j int) bool {
+		return ret[i] < ret[j]
+	})
+	return ret
```

Comment:
Why not return a slice of column elements here instead? We usually prefer manipulating objects directly, rather than references to objects, whenever possible.

Also I'm not sure whether you need to sort them at all. The order in which they're in should be stable, I think.

Reply:
I've changed the return to `[]*scpb.Column` and got rid of the sorting step.